### PR TITLE
Engagement/nj prework

### LINKED ISSUE #13855: New Jersey - UP Receiver migration Pre-work
https://cdc.sharepoint.com/:w:/r/teams/ReportStream/_layouts/15/Doc.aspx?sourcedoc=%7B1f9d0372-366a-4041-acc7-38f5f8cd1025%7D&action=edit&wdPreviousSession=68200238-341d-38b4-94a3-49a6d5c3ad9b

### User Story:
As a developer, I want to compare the messages generated from the Covid and Universal pipelines, so that I can successfully onboard [STLT] to the UP

### Description/Use Case
[STLT] is already receiving data from the covid pipeline they will be onboarded to the UP where they will be receiving data from SimpleReport including Covid, Flu and RSV. We need to make sure the message we generate from the UP matches the Covid pipeline message.

### Dev Notes:

- [ ] Fetch [STLT] organization settings from production and load them locally
- [ ] Use the attached SimpleReport covid postman collection and make sure the message gets routed to [STLT] locally. Modify the message to meet [STLT] filter if needed [Simple Report Covid.postman_collection](https://github.com/CDCgov/prime-reportstream/blob/master/prime-router/docs/onboarding-users/samples/SimpleReport/Simple%20Report%20Covid.postman_collection.json)
- [ ] Make a copy of the [STLT] organization settings to onboard them to the UP. See How to Migrate an existing receiver to the UP documentation for more details: https://github.com/CDCgov/prime-reportstream/blob/master/prime-router/docs/onboarding-users/migrating-receivers.md
- [ ] Use this Postman collection to send a FHIR bundle the UP and make sure the message gets routed to the new UP [STLT] receiver. You may need to update the Simple Report sender to use the simple-report-sender-transform.yml if it's not using it. [Simple Report UP.postman_collection](https://github.com/CDCgov/prime-reportstream/blob/master/prime-router/docs/onboarding-users/samples/SimpleReport/Simple%20Report%20UP.postman_collection.json)
- To migrate the Covid translation settings start by looking at their current translation settings. If the receiver uses  any of the following settings you will need to create a receiver schema:
	- receivingApplicationName
	- receivingApplicationOID
	- receivingFacilityName
	- receivingFacilityOID
	- messageProfileId
	- replaceValue
	- replaceValueAwithB
	- reportingFacilityName
	- reportingFacilityId
    - reportingFacilityIdType
    - suppressQstForAoe: true
    - suppressHl7Fields
    - suppressAoe: true
    - defaultAoeToUnknown
    - replaceUnicodeWithAscii
    - useBlankInsteadOfUnknown
    - usePid14ForPatientEmail: true
    - suppressNonNPI
    - replaceDiiWithOid
    - useOrderingFacilityName not STANDARD
    - receivingOrganization
    - stripInvalidCharsRegex
    
- More documentation on how to set-up these transforms in the UP will be provided, but for now you can look for examples on how to set this up in either the NY-receiver-transforms or CA-receiver-transforms

- If the receiver uses any of those transforms you will need to create a receiver transform under `metadata/hl7_mapping/receivers/STLTs/` and update the receiver settings to point to this schema.

- After migrating the receiver setting to the UP. Send another test message using the postman collection or feel free to use the PrimeCLI to test the sender and receiver transforms using this command:
`./prime fhirdata --input-file {PATH TO INPUT FILE} -s metadata/fhir_transforms/senders/SimpleReport/simple-report-sender-transform.yml --output-format HL7 -r metadata/hl7_mapping/receivers/STLTs/{RECEIVER SCHEMA} --output-file {PATH TO OUTPUT FILE}`

- Compare the two messages again and document if there are any differences between the two messages. If there are review them with the team. We will probably need to ask the sender to add missing data or add a sender transform.

- [ ] Use a diff tool of your choice and compare the two messages generated between the two pipelines and document the differences. You can find an example here [RI/SR UP vs Covid](https://docs.google.com/spreadsheets/d/197AeFMvozqUGRE1BuvOSMiUL_r2EEkyQv4l8D_OhhZk/edit#gid=492389121)

- If there are no major differences we can move on to sending test messages to the STLTs staging environment.

### Acceptance Criteria 
- [ ] Created and sent data to [STLT] through the covid pipeline locally
- [ ] Created and sent data to [STLTS] through the universal pipeline locally
- [ ] Migrated  Covid receiver translation settings to the UP receiver settings
- [ ] Successfully generated a message with migrated UP receiver settings
- [ ] Review transforms settings with the team
- [ ] Compared messages from the covid and universal pipelines and documented differences and review with team

### PR DESCRIPTION
**Pull request by victor-chaparro**
This PR adds NJ pre-work and integration test for testing NJ receiver transforms and receiver settings.

Test Steps:
1. Run `./gradlew testIntegration`
2. Run `./prime multiple-settings set --input ./settings/STLTs/NJ/nj-doh.yml --env local`
3. Send a report routed to NJ
4. Verify report gets routed to NJ

## Changes
- Created transform for NJ
- Included settings for new UP receiver for NJ

### Testing
- [X] Tested locally?
- [X] Ran `./prime test` or `./gradlew testSmoke` against local Docker ReportStream container? (didn't change 
- [ ] (For Changes to /frontend-react/...) Ran `npm run lint:write`? 
- [X] Added tests?

## Linked Issues
- Fixes #13855 

## To Be Done
*Create GitHub issues to track the work remaining, if any*
- #14264 

image

### CODE REVIEW

File: prime-router/settings/STLTs/NJ/nj-doh.yml [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+  receivers:
+  - name: "full-elr"
+    organizationName: "nj-doh"
+    topic: "full-elr"
+    customerStatus: "active"
+    translation: !<HL7>
+      schemaName: "azure:/hl7_mapping/receivers/STLTs/NJ/NJ-receiver-transform.yml"
+      useTestProcessingMode: false
+      useBatchHeaders: true
+      receivingApplicationName: null
```
Comment (JessicaWNava) [inline prime-router/settings/STLTs/NJ/nj-doh.yml:19]: This should be ELRS

File: prime-router/settings/STLTs/NJ/nj-doh.yml [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+      receivingApplicationOID: null
+      receivingFacilityName: null
```
Comment (JessicaWNava) [inline prime-router/settings/STLTs/NJ/nj-doh.yml:21]: This should be NJDOH

File: prime-router/settings/STLTs/NJ/nj-doh.yml [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+  - name: "full-elr"
+    organizationName: "nj-doh"
+    topic: "full-elr"
+    customerStatus: "active"
+    translation: !<HL7>
+      schemaName: "azure:/hl7_mapping/receivers/STLTs/NJ/NJ-receiver-transform.yml"
+      useTestProcessingMode: false
+      useBatchHeaders: true
+      receivingApplicationName: null
+      receivingApplicationOID: null
```
Comment (JessicaWNava) [inline prime-router/settings/STLTs/NJ/nj-doh.yml:20]: This should be 2.16.840.1.113883.3.1299.5.1.6.1

File: prime-router/settings/STLTs/NJ/nj-doh.yml [hunk 4]
```diff
@@ -0,0 +1,110 @@
...
+      receivingFacilityName: null
+      receivingFacilityOID: null
```
Comment (JessicaWNava) [inline prime-router/settings/STLTs/NJ/nj-doh.yml:22]: This should be 2.16.840.1.113883.3.1299

File: prime-router/src/main/resources/metadata/hl7_mapping/receivers/STLTs/NJ/NJ-receiver-transform.yml [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+    value: [ '"NJDOH"' ]
+    hl7Spec: [ MSH-6-1 ]
+
+  - name: nj-receiving-facility-universal-id
+    value: [ '"2.16.840.1.113883.3.1299"' ]
+    hl7Spec: [ MSH-6-2 ]
+
+  - name: nj-receiving-facility-universal-id-type
+    value: [ '"ISO"' ]
+    hl7Spec: [ MSH-6-3 ]
```
Comment (JessicaWNava) [inline prime-router/src/main/resources/metadata/hl7_mapping/receivers/STLTs/NJ/NJ-receiver-transform.yml:59]: These fields need to be suppressed: OBX-18-1,OBX-18-2,OBX-18-3

Comment (JessicaWNava) [inline prime-router/src/main/resources/metadata/hl7_mapping/receivers/STLTs/NJ/NJ-receiver-transform.yml:59]: Was truncateHDNamespaceIds: true taken into account here?

**Review by JessicaWNava — CHANGES_REQUESTED** [review]
Few changes but looks really good!

File: prime-router/src/main/resources/metadata/fhir_transforms/common/suppress-device-information.yml [hunk 6]
```diff
@@ -0,0 +1,28 @@
+## Suppresses Device information OBX-18-1,OBX-18-2,OBX-18-3
```
Comment (GilmoreA6) [inline prime-router/src/main/resources/metadata/fhir_transforms/common/suppress-device-information.yml:1]: These can all be suppressed by overiding element "obx-equipment-instance-identifier" in the mappings

  - name: obx-equipment-instance-identifier
    condition: false

Comment (victor-chaparro) [reply prime-router/src/main/resources/metadata/fhir_transforms/common/suppress-device-information.yml:1]: That does make it a lot easier. I was also trying to start using enrichment schemas for stuff like that, but this is a good example of how much more complicated transforms are in FHIR to FHIR.

**Review by JessicaWNava — APPROVED** [review]
LGTM thanks for making those changes!

**Review by GilmoreA6 — APPROVED** [review]

**Pull request merged**